New socket close() implementation broke build of platforms using socket offloading

**Describe the bug**
Errors are seen when building examples for cc3220sf_launchxl:

```
[90/172] Building C object zephyr/CMak...fi/simplelink/simplelink_sockets.c.obj
FAILED: zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj
ccache /home/local/bin/zephyr-sdk-0.11.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DBUILD_VERSION=zephyr-v2.3.0-1876-g819f4e00e6d1 -DKERNEL -DSL_PLATFORM_MULTI_THREADED -DSL_SUPPORT_IPV6 -DUSE_CC3220_ROM_DRV_API -D_FORTIFY_SOURCE=2 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR__=1 -I../../../../../include -Izephyr/include/generated -I../../../../../soc/arm/ti_simplelink/cc32xx -I../../../../../lib/libc/newlib/include -I../../../../../subsys/net/lib/sockets/. -I../../../../../drivers/wifi/simplelink/. -I../../../../../subsys/net/lib/tls_credentials -I/home/local/zephyrproject-work/modules/hal/cmsis/CMSIS/Core/Include -I/home/local/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/. -I/home/local/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/inc -I/home/local/zephyrproject-work/modules/hal/ti/simplelink/source/ti/devices/cc32xx/driverlib -I/home/local/zephyrproject-work/modules/hal/ti/simplelink/. -I/home/local/zephyrproject-work/modules/hal/ti/simplelink/source -I/home/local/zephyrproject-work/modules/hal/ti/simplelink/kernel/zephyr/dpl -I/home/local/zephyrproject-work/modules/hal/ti/simplelink/source/ti/drivers/net/wifi/porting -Og -imacros/home/local/zephyrproject-work/zephyr/samples/net/sockets/http_get/build/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -mcpu=cortex-m4 -mthumb -mabi=aapcs -imacros/home/local/zephyrproject-work/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-address-of-packed-member -Wno-pointer-sign -Wpointer-arith -Wno-unused-but-set-variable -Werror=implicit-int -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fno-reorder-functions -fno-defer-pop -fmacro-prefix-map=/home/local/zephyrproject-work/zephyr/samples/net/sockets/http_get=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/local/zephyrproject-work/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/local/zephyrproject-work=WEST_TOPDIR -ffunction-sections -fdata-sections -specs=nano.specs -std=c99 -MD -MT zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj -MF zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj.d -o zephyr/CMakeFiles/zephyr.dir/drivers/wifi/simplelink/simplelink_sockets.c.obj   -c /home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:1127:12: error: conflicting types for 'simplelink_close'
 1127 | static int simplelink_close(void *obj)
      |            ^~~~~~~~~~~~~~~~
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:284:12: note: previous definition of 'simplelink_close' was here
  284 | static int simplelink_close(int sd)
      |            ^~~~~~~~~~~~~~~~
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c: In function 'simplelink_close':
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:1129:26: error: 'sd' undeclared (first use in this function); did you mean 'send'?
 1129 |  return simplelink_close(sd);
      |                          ^~
      |                          send
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:1129:26: note: each undeclared identifier is reported only once for each function it appears in
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:1130:1: warning: control reaches end of non-void function [-Wreturn-type]
 1130 | }
      | ^
At top level:
/home/local/zephyrproject-work/zephyr/drivers/wifi/simplelink/simplelink_sockets.c:284:12: warning: 'simplelink_close' defined but not used [-Wunused-function]
  284 | static int simplelink_close(int sd)
      |            ^~~~~~~~~~~~~~~~
[95/172] Linking C static library zeph...xl/libboards__arm__cc3220sf_launchxl.a
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/local/bin/cmake/cmake-3.13.3-Linux-x86_64/bin/cmake --build /home/local/zephyrproject-work/zephyr/samples/net/sockets/http_get/build
```


This is due to changes in recent PR #27485. Looking at the changes, I believe the build should be broken for eswifi and ublox-sara-r4 modem as well.

**To Reproduce**
Steps to reproduce the behavior:
1. cd samples/net/sockets/http_get
2. west build -b cc3220sf_launchxl
3. See error

**Expected behavior**
The examples should build correctly.

**Impact**
Cannot build examples that use socket offloading on affected platforms.

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - Toolchain (e.g Zephyr SDK, ...): Zephyr SDK 0.11.3
 - Commit SHA or Version used: 819f4e00e6d11acd3ad40d4e98bf308cb6e10060

--- a/drivers/modem/ublox-sara-r4.c
+++ b/drivers/modem/ublox-sara-r4.c
@@ -1205,8 +1205,9 @@ static int offload_socket(int family, int type, int proto)
 	return ret;
 }
 
-static int offload_close(struct modem_socket *sock)
+static int offload_close(void *obj)
 {
+	struct modem_socket *sock = (struct modem_socket *)obj;
 	char buf[sizeof("AT+USOCL=#\r")];
 	int ret;
 
@@ -1451,11 +1452,6 @@ static ssize_t offload_sendto(void *obj, const void *buf, size_t len,
 
 	errno = 0;
 	return ret;
-}
-
-static int offload_close(void *obj)
-{
-	return offload_close((struct modem_socket *)obj);
 }
 
 static int offload_ioctl(void *obj, unsigned int request, va_list args)

--- a/drivers/wifi/eswifi/eswifi_socket_offload.c
+++ b/drivers/wifi/eswifi/eswifi_socket_offload.c
@@ -350,8 +350,9 @@ static ssize_t eswifi_socket_recvfrom(void *obj, void *buf, size_t len,
 	return eswifi_socket_recv(obj, buf, len, flags);
 }
 
-static int eswifi_socket_close(int sock)
+static int eswifi_socket_close(void *obj)
 {
+	int sock = OBJ_TO_SD(obj);
 	struct eswifi_off_socket *socket;
 	struct net_pkt *pkt;
 	int ret;
@@ -497,11 +498,6 @@ static int eswifi_socket_create(int family, int type, int proto)
 					&eswifi_socket_fd_op_vtable);
 
 	return fd;
-}
-
-static int eswifi_socket_close(void *obj)
-{
-	return eswifi_socket_close(sd);
 }
 
 static int eswifi_socket_ioctl(void *obj, unsigned int request, va_list args)

--- a/drivers/wifi/simplelink/simplelink_sockets.c
+++ b/drivers/wifi/simplelink/simplelink_sockets.c
@@ -281,8 +281,9 @@ exit:
 	return retval;
 }
 
-static int simplelink_close(int sd)
+static int simplelink_close(void *obj)
 {
+	int sd = OBJ_TO_SD(obj);
 	int retval;
 
 	retval = sl_Close(sd);
@@ -1122,11 +1123,6 @@ static int simplelink_fcntl(int sd, int cmd, va_list args)
 
 exit:
 	return retval;
-}
-
-static int simplelink_close(void *obj)
-{
-	return simplelink_close(sd);
 }
 
 static int simplelink_ioctl(void *obj, unsigned int request, va_list args)

